src/org/zalando/stups/even/ssh.clj

```diff
@@ -23,27 +26,39 @@
 (defn write-key-to-file
   "Write private SSH key to a temp file, only readable by our user"
   [key]
-  (let [path (str (Files/createTempFile "ssh-private-key" "sshkey.pem" owner-only))]
+  (let [key-filename (format "%s.pem" (UUID/randomUUID))
+        path (str (Files/createTempFile "ssh-private-key" key-filename owner-only))]
     (log/info "Writing SSH private key to" path)
     (spit path key)
     path))
 
-(defn get-private-key-path
-  "Return path to the private key written to disk if key is the actual key (PEM encoded)"
-  [key]
-  (if (.startsWith key "-----BEGIN") (write-key-to-file key) key))
+(defn split-keys
+  "Extract all keys contained in a single string into a sequence of keys"
+  [key-str]
+  (->> (str/split key-str #"(?=-----BEGIN)")
+    (map str/trim)
+    (filter not-empty)))
+
+(defn write-private-keys
+  "Takes a string containing multiple private keys, writes each of them to an individual file and returns
+   a sequence of paths to these files"
+  [key-str]
+  (map write-key-to-file (split-keys key-str)))
 
 (defn set-timeout [session timeout]
   (.setTimeout session timeout))
 
 (defn execute-ssh
   "Execute the given command on the remote host using the configured SSH user and private key"
-  [hostname command {{:keys [user private-key port agent-forwarding timeout]} :config}]
+  [hostname command {{:keys [user private-keys private-key-password port agent-forwarding timeout]} :config}]
   (log/info "ssh" user "@" hostname command)
   (let [agent (ssh-agent {:use-system-ssh-agent false
                           :known-hosts-path     "/dev/null"})
-        private-key-path (get-private-key-path private-key)]
-    (add-identity agent {:private-key-path private-key-path})
+        private-key-paths (write-private-keys private-keys)]
+    (doseq [path private-key-paths]
+      (add-identity agent {:private-key-path path
+                           :passphrase       (some->> private-key-password
```

Comment:
Shouldn't this one be a single arrow `some->`, because `.getBytes` takes the string as **first** argument?